### PR TITLE
rewrote docs to better illustrate synergy bw sim and real

### PR DESCRIPTION
**Pull request by shrutij01**
## Description

rewrote docs for better design flow (and fixed some logical issues), renamed gym envs to reflect the TriFinger and not just Finger

## How I Tested

by running all the unit tests

- [x] All new code is formatted according to our style guide (for C++ run clang-format, for Python, run flake8 and fix all warnings).
- [x] All new functions/classes are documented and existing documentation is updated according to changes.
- [x] No commented code from testing/debugging is kept (unless there is a good reason to keep it).

### CODE REVIEW

File: docs/simreal/simvsreal.rst [hunk 1]
```diff
@@ -1,29 +1,16 @@
...
-  ``SimFinger`` is available (so you won't have to keep the below
-  differences in mind betwen the simulation and the real robot). 
-  However, a drawback of using this approach is that this way it is
-  not possible to directly access the simulation, e.g. to reset the finger.
-  For more details see :ref:`robot_interfaces with Simulation`.
+The ``SimFinger`` class provides us with an API to control the TriFinger robots in
+simulation. Similarly, the `robot_interfaces::RobotFrontend`_ class provides an
+API to interact with these robots in real life.
 
+It is extremely easy to switch between these two. However, there are some subtle
```
Comment (luator) [inline docs/simreal/simvsreal.rst:11]: Maybe a matter of taste but I would be a bit careful with words like "extremely".  I think "It is easy to switch" would be enough here.

File: docs/simreal/simwithreal.rst [hunk 2]
```diff
@@ -0,0 +1,64 @@
+.. _`Switching between the Real Robot and the Simulated Robot in this Simulation`:
```
Comment (luator) [inline docs/simreal/simwithreal.rst:1]: This label is a bit long.  Actually, you don't really need labels at the top of the file.  If you want to reference the whole page, you can use

    :doc:`filename`

File: docs/simreal/simwithreal.rst [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+*****************************************************************************
+Switching between the Real Robot and the Simulated Robot in this Simulation
```
Comment (luator) [inline docs/simreal/simwithreal.rst:4]: The title is a bit confusing (we are not using the real robot _in_ the simulation).
Also, given that I consider the `RealRobot` class deprecated, I'm not sure if we should even mention in the documentation.

File: docs/index.rst [hunk 4]
```diff
@@ -22,24 +22,25 @@ Dive in!
...
 
 .. toctree::
-   :maxdepth: 2
+   :maxdepth: 1
    :caption: Ease of Use with a Real Robot
    
-   Simulation vs Real Robot <simvsreal>
-   Interface with a Real Robot <driverinterface>
-   Timestep Logic <timesteplogic>
+   Switching between the Real Robot and the Simulated Robot in this Simulation <simreal/simwithreal>
```
Comment (luator) [inline docs/index.rst:33]: From what I saw, the titles here are the same as the titles of the actual files.  In this case, you don't need to repeat it here but you can simply list the filename.

```suggestion
   simreal/simwithreal
```

File: docs/simreal/timesteplogic.rst [hunk 5]
```diff
@@ -1,6 +1,6 @@
-***************************
-Timestep Logic
-***************************
+********************************
+"Real" and "Non-real" TimeSteps
```
Comment (luator) [inline docs/simreal/timesteplogic.rst:2]:
```suggestion
"Real" and "Non-real" Time Steps
```

Comment (luator) [reply docs/simreal/timesteplogic.rst:2]: This title is not very clear to me.  What would be a non-real timestep?  Do you mean non-real-time?

File: docs/simreal/simvsreal.rst [hunk 6]
```diff
@@ -1,29 +1,16 @@
...
-simulation. And, how can these robots be controlled in real life? via the
-`robot_interfaces::RobotFrontend`_ class.
-
-We aim to have the same API for controlling the robot in the ``SimFinger`` class
-as in ``robot_interfaces::RobotFrontend`` to allow transition from simulation to
-real robot with only minimal modifications to the code. There are still a few
-important differences one should keep in mind though. These differences are
-described below:
+********************************************************
+Switching between Non-real-time and Real-time Behaviour
```
[the remark this answers is not in the document]
Comment (shrutij01) [reply docs/simreal/simvsreal.rst:4]: Yeah, I agree. But I think that doesn't flow well with the entire theme of showing how it is very easy to use both the simulated and the real robots through our interface... so I removed explicit attention via a "vs" comparison between the two.

File: docs/simreal/simvsreal.rst (hunk 1 above)
Comment (shrutij01) [reply docs/simreal/simvsreal.rst:11]: Yeah, I think it's my writing style: I tend to use more adjectives and adverbs. It is, in fact, extremely easy to use :P So we could make that explicit like this

File: docs/simreal/simwithreal.rst (hunk 2 above)
Comment (shrutij01) [reply docs/simreal/simwithreal.rst:1]: thank you :)

File: docs/simreal/timesteplogic.rst (hunk 5 above)
Comment (shrutij01) [reply docs/simreal/timesteplogic.rst:2]: I think it sounds good... makes it clear the notion of time is different in simulation and reality, and then this section goes on to explain that

File: docs/index.rst [hunk 7]
```diff
@@ -30,10 +30,10 @@ Dive in!
    :maxdepth: 1
    :caption: Ease of Use with a Real Robot
    
-   Switching between the Real Robot and the Simulated Robot in this Simulation <simreal/simwithreal>
-   Switching between the Real Robot and the Simulated Robot in the Real Robot API <simreal/realwithsim>
-   Switching between Non-real-time and Real-time Behaviour <simreal/simvsreal>
-   "Real" and "Non-real" TimeSteps <simreal/timesteplogic>
+   simreal/simwithreal>
```
Comment (luator) [inline docs/index.rst]: trailing `>`

File: docs/simreal/simwithreal.rst [hunk 8]
```diff
@@ -0,0 +1,62 @@
+**********************************************************************************
```
Comment (luator) [inline docs/simreal/simwithreal.rst:1]: We should probably mention somewhere on this page that `RealRobot` depends on other packages which are not part of the conda environment.

File: docs/index.rst (hunk 7 above)
Comment (shrutij01) [reply docs/index.rst]: oh, this is an old change, i have already fixed this

File: docs/simreal/simwithreal.rst (hunk 8 above)
Comment (shrutij01) [reply docs/simreal/simwithreal.rst:1]: ah right, that's very important

**Comment by shrutij01** [comment]
thank you :)

**Pull request merged**